cli/tests/pcluster/api/controllers/test_cluster_instances_controller.py

```diff
@@ -5,43 +5,260 @@
+            actual_filters_dict = {}
+            for filter in actual_filters:
+                actual_filters_dict[filter["Name"]] = filter["Values"][0]
+            assert_that(actual_filters_dict.get("tag:parallelcluster:cluster-name")).is_equal_to(cluster_name)
+            if node_type:
+                if node_type == NodeType.HEAD:
+                    expected_value = "HeadNode"
+                else:
+                    expected_value = "Compute"
+            else:
+                expected_value = None
+            assert_that(actual_filters_dict.get("tag:parallelcluster:node-type")).is_equal_to(expected_value)
+            assert_that(actual_filters_dict.get("tag:QueueName")).is_equal_to(queue_name)
+
+    @pytest.mark.parametrize(
+        "params, expected_response",
+        [
+            (
+                {"node_type": "wrong_node_type"},
+                {"message": "Bad Request: 'wrong_node_type' is not one of ['HEAD', 'COMPUTE']"},
+            ),
+            (
+                {"region": "eu-west-"},
+                {"message": "Bad Request: invalid or unsupported region 'eu-west-'"},
+            ),
+            (
+                {"region": None},
+                {"message": "Bad Request: region needs to be set"},
+            ),
+        ],
+    )
+    def test_malformed_request(self, client, params, expected_response):
+        response = self._send_test_request(client, **params)
+        with soft_assertions():
+            assert_that(response.status_code).is_equal_to(400)
+            assert_that(response.get_json()).is_equal_to(expected_response)
+
+
+def cfn_describe_stack_mock_response(scheduler="slurm", version="3.0.0"):
+    stack_data = {
+        "StackName": "clustername",
+        "Parameters": [{"ParameterKey": "Scheduler", "ParameterValue": scheduler}],
+        "Tags": [{"Key": "parallelcluster:version", "Value": version}],
+    }
+    return stack_data
+
+
+class TestDeleteClusterInstances:
+    url = "/v3/clusters/{cluster_name}/instances"
+    method = "DELETE"
+
+    def _send_test_request(self, client, cluster_name="clustername", region="us-east-1", force=False):
+        query_string = []
+        if region:
+            query_string.append(("region", region))
+        if force:
+            query_string.append(("force", force))
+
         headers = {
             "Accept": "application/json",
         }
-        response = client.open(
-            "/v3/clusters/{cluster_name}/instances".format(cluster_name="clustername"),
-            method="GET",
-            headers=headers,
-            query_string=query_string,
+        return client.open(
+            self.url.format(cluster_name=cluster_name), method=self.method, headers=headers, query_string=query_string
+        )
+
+    @pytest.mark.parametrize("stack_exists, force", [(True, False), (False, True), (True, True)])
+    def test_successful_request(self, mocker, client, stack_exists, force):
```

Comment:
can you add a test for next_token? it can also be a separate test if it keeps it simpler

Reply:
This is the test for delete cluster instances, where next token does not exist. I already added a next token test for describe cluster instances.

Reply:
whoops sorry :)